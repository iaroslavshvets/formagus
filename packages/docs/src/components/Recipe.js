import React from 'react';

export const Recipe = ({sandboxId}) => (
  <>
    <iframe
      id="recipe"
      title="formagus recipe"
      src={`https://codesandbox.io/embed/${sandboxId}?fontsize=13&module=/src/ExampleForm.tsx&view=split`}
      style={{
        display: 'block',
        width: '100%',
        border: 0,
        height: '100vh',
      }}
      sandbox="allow-modals allow-forms allow-popups allow-scripts allow-same-origin"
    />
  </>
);
