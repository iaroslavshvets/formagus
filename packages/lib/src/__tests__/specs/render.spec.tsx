import {cleanup, render} from '@testing-library/react';
import React from 'react';
import {TestForm} from '../components/TestForm';
import {createTestFormDriver} from '../components/createTestFormDriver';
import {Field} from '../../index';
import {Input} from '../components/Input';

describe('Render', () => {
  afterEach(() => {
    cleanup();
  });

  it('should render', () => {
    const wrapper = render(
      <TestForm>
        <Field
          name={TestForm.FIELD_ONE_NAME}
          render={(props) => {
            return <Input {...props} useHook={false} />;
          }}
        />
        <Field name={TestForm.FIELD_NESTED_NAME}>
          <Input />
        </Field>
      </TestForm>,
    ).container;

    const formDriver = createTestFormDriver({wrapper});

    expect(formDriver.get.serialized()).toMatchSnapshot();
  });
});
