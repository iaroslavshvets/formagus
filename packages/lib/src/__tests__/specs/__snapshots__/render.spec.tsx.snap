// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Render > should render 1`] = `
<div>
  <form
    data-hook="test-form"
    novalidate=""
  >
    <div
      data-hook="field_one_name"
    >
      <input
        data-hook="input-field_one_name"
        name="field_one_name"
        value=""
      />
      <div
        data-hook="state-props"
      >
        <span
          data-hook="field_state_initialValue"
        >
          undefined
        </span>
        <span
          data-hook="field_state_isActive"
        >
          false
        </span>
        <span
          data-hook="field_state_isDirty"
        >
          false
        </span>
        <span
          data-hook="field_state_isTouched"
        >
          false
        </span>
        <span
          data-hook="field_state_isChanged"
        >
          false
        </span>
        <span
          data-hook="field_state_isValidating"
        >
          false
        </span>
        <span
          data-hook="field_state_isMounted"
        >
          true
        </span>
        <span
          data-hook="form_state_isValidating"
        >
          false
        </span>
        <span
          data-hook="form_state_isSubmitting"
        >
          false
        </span>
        <span
          data-hook="form_state_submitCount"
        >
          0
        </span>
        <span
          data-hook="form_state_isValid"
        >
          true
        </span>
        <span
          data-hook="form_state_isDirty"
        >
          false
        </span>
        <span
          data-hook="form_state_isTouched"
        >
          false
        </span>
        <span
          data-hook="form_state_isChanged"
        >
          false
        </span>
      </div>
      <span
        data-hook="validate-field"
      />
      <span
        data-hook="validate"
      />
    </div>
    <div
      data-hook="root.field.nested"
    >
      <input
        data-hook="input-root.field.nested"
        name="root.field.nested"
        value=""
      />
      <div
        data-hook="state-props"
      >
        <span
          data-hook="field_state_initialValue"
        >
          undefined
        </span>
        <span
          data-hook="field_state_isActive"
        >
          false
        </span>
        <span
          data-hook="field_state_isDirty"
        >
          false
        </span>
        <span
          data-hook="field_state_isTouched"
        >
          false
        </span>
        <span
          data-hook="field_state_isChanged"
        >
          false
        </span>
        <span
          data-hook="field_state_isValidating"
        >
          false
        </span>
        <span
          data-hook="field_state_isMounted"
        >
          true
        </span>
        <span
          data-hook="form_state_isValidating"
        >
          false
        </span>
        <span
          data-hook="form_state_isSubmitting"
        >
          false
        </span>
        <span
          data-hook="form_state_submitCount"
        >
          0
        </span>
        <span
          data-hook="form_state_isValid"
        >
          true
        </span>
        <span
          data-hook="form_state_isDirty"
        >
          false
        </span>
        <span
          data-hook="form_state_isTouched"
        >
          false
        </span>
        <span
          data-hook="form_state_isChanged"
        >
          false
        </span>
      </div>
      <span
        data-hook="validate-field"
      />
      <span
        data-hook="validate"
      />
    </div>
    <div
      data-hook="form-values"
    >
      {"root":{"field":{}}}
    </div>
    <button
      type="submit"
    >
      Submit
    </button>
  </form>
</div>
`;
